import.go

```diff
@@ -28,12 +30,39 @@ func importBuilderDataFromImage(ctx context.Context, store storage.Store, system
 	if err != nil {
 		return nil, errors.Wrapf(err, "no such image %q", imageID)
 	}
-	src, err2 := ref.NewImage(ctx, systemContext)
-	if err2 != nil {
-		return nil, errors.Wrapf(err2, "error instantiating image")
+	src, err := ref.NewImageSource(ctx, systemContext)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error instantiating image source")
 	}
 	defer src.Close()
 
+	imageDigest := ""
+	manifestBytes, manifestType, err := src.GetManifest(ctx, nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error loading image manifest for %q", transports.ImageName(ref))
+	}
+	if manifestDigest, err := manifest.Digest(manifestBytes); err == nil {
+		imageDigest = manifestDigest.String()
+	}
+
+	var instanceDigest *digest.Digest
+	if manifest.MIMETypeIsMultiImage(manifestType) {
+		list, err := manifest.ListFromBlob(manifestBytes, manifestType)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error parsing image manifest for %q as list", transports.ImageName(ref))
+		}
+		instance, err := list.ChooseInstance(systemContext)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error finding an appropriate image in manifest list %q", transports.ImageName(ref))
+		}
+		instanceDigest = &instance
+	}
+
+	img, err := image.FromUnparsedImage(ctx, systemContext, image.UnparsedInstance(src, instanceDigest))
```

Comment:
Your returning an img and then three lines later replacing it?

Reply:
Not sure I follow.  This is mainly about making sure that `img` is pointing to a runnable image if the reference we started with included the digest of a manifest list or image index.  If it isn't a list, then `instanceDigest` is `nil` and we basically continue as we would have otherwise, but if it is, we use the `SystemContext` to choose an instance and attempt to use that one, instead.

Reply:
You are not using the value returned above, then you replace it below.
It would be clearer if you did 
if _, err := image.FromUnparsedImage(ctx, systemContext, image.UnparsedInstance(src, instanceDigest)); err != nil

Reply:
Ah.  That's a different variable, of a different type.  Renamed the new one.